packages/animated-qr/src/AnimatedQRScanner/getAnimatedScan.ts

```diff
@@ -26,7 +27,14 @@ export const getAnimatedScan = ({
 
       if (urDecoder.isSuccess()) {
         const ur = urDecoder.resultUR();
-        if (purposeToURType[purpose].includes(ur.type)) {
+        const types = []
+        if (purpose && purposeToURType[purpose]) {
+          types.push(...purposeToURType[purpose])
+        }
+        if (urTypes instanceof Array) {
```

Comment:
I feel like we don't need to check type here, what about giving `urTypes` a default value `[]`